File: Tema 3 -Sentencia condicional (if-switch)/ejercicios/ejercicio10 horoscopo a partir del dia y mes de nacimiento/archives/10.py
import sys
import math

# Obtener los argumentos de la línea de comandos
dia = int(sys.argv[1])
mes = int(sys.argv[2])

if ((dia > 0 and dia <= 31) and (mes > 0 and mes <= 12)) and (
    (mes == 2 and dia <= 29)
    or ((mes == 4 or mes == 6 or mes == 9 or mes == 11) and dia <= 30)
    or (
        (
            mes == 1
            or mes == 3
            or mes == 5
            or mes == 7
            or mes == 8
            or mes == 10
            or mes == 12
        )
        and dia <= 31
    )
):
    if (dia >= 21 and mes == 3) or (dia <= 19 and mes == 4):
        mensaje = f"""Su fecha de nacimiento es: <br> {dia} de {("Marzo"if mes == 3 else"Abril")} <br> Su signo es Aries"""

    elif (dia >= 20 and mes == 4) or (dia <= 20 and mes == 5):
        mensaje = f"""Su fecha de nacimiento es: <br> {dia} de {("Abril" if mes == 4 else "Mayo")} <br> Su signo es Tauro"""

    elif (dia >= 21 and mes == 5) or (dia <= 20 and mes == 6):
        mensaje = f"""Su fecha de nacimiento es: <br> {dia} de {("Mayo"if mes == 5 else "Junio")} <br> Su signo es Geminis"""

    elif (dia >= 21 and mes == 6) or (dia <= 22 and mes == 7):
        mensaje = f"""Su fecha de nacimiento es: <br> {dia} de {("Junio"if mes == 6 else "Julio")} <br> Su signo Cancer"""

    elif (dia >= 23 and mes == 7) or (dia <= 22 and mes == 8):
        mensaje = f"""Su fecha de nacimiento es: <br> {dia} de {("Julio"if mes == 7 else "Agosto")} <br> Su signo es Leo"""

    elif (dia >= 23 and mes == 8) or (dia <= 22 and mes == 9):
        mensaje = f"""Su fecha de nacimiento es: <br> {dia} de {("Agosto"if mes == 8 else "Septiembre")} <br> Su signo es Virgo"""

    elif (dia >= 23 and mes == 9) or (dia <= 22 and mes == 10):
        mensaje = f"""Su fecha de nacimiento es: <br> {dia} de {("Septiembre"if mes == 9 else "Octubre")} <br> Su signo es Libra"""

    elif (dia >= 23 and mes == 10) or (dia <= 21 and mes == 11):
        mensaje = f"""Su fecha de nacimiento es: <br> {dia} de {("Octubre"if mes == 10 else "Noviembre")} <br> Su signo es Escorpio"""

    elif (dia >= 22 and mes == 11) or (dia <= 21 and mes == 12):
        mensaje = f"""Su fecha de nacimiento es: <br> {dia} de {("Noviembre"if mes == 11 else "Diciembre")} <br> Su signo es Sagitario"""

    elif (dia >= 22 and mes == 12) or (dia <= 19 and mes == 1):
        mensaje = f"""Su fecha de nacimiento es: <br> {dia} de {("Diciembre"if mes == 12 else "Enero")} <br> Su signo es Capricornio"""

    elif (dia >= 20 and mes == 1) or (dia <= 18 and mes == 2):
        mensaje = f"""Su fecha de nacimiento es: <br> {dia} de {("Enero"if mes == 1 else "Febrero")} <br> Su signo es Acuario"""

    elif (dia >= 19 and mes == 2) or (dia <= 20 and mes == 3):
        mensaje = f"""Su fecha de nacimiento es: <br> {dia} de {("Febrero"if mes == 2 else "Marzo")} <br> Su signo es Piscis"""
else:
    mensaje = f"Verifique la fecha de su nacimiento"

print(f"{mensaje}")

File: Tema 3 -Sentencia condicional (if-switch)/ejercicios/ejercicio10 horoscopo a partir del dia y mes de nacimiento/archives/10.js
function sumarJS() {
  const dia = parseInt(document.getElementById("num1").value);
  const mes = parseInt(document.getElementById("num2").value);
  if (Number.isFinite(dia) && Number.isFinite(mes)) {
    /* SECTION inicio codigo tras verificacion de formulario */
    if (((dia > 0 && dia <= 31) && (mes > 0 && mes <= 12)) && ((mes == 2 && dia <= 29) || ((mes == 4 || mes == 6 || mes == 9 || mes == 11) && dia <= 30) || ((mes == 1 || mes == 3 || mes == 5 || mes == 7 || mes == 8 || mes == 10 || mes == 12) && dia <= 31))) {

      if ((dia >= 21 && mes == 3) || (dia <= 19 && mes == 4)) {

        mensaje = `Su fecha de nacimiento es:<br> ${dia} de ${(mes == 3 ? "Marzo" : "Abril")}<br>Su signo es Aries`;

      } else if ((dia >= 20 && mes == 4) || (dia <= 20 && mes == 5)) {
        mensaje = `Su fecha de nacimiento es:<br> ${dia} de ${(mes == 4 ? "Abril" : "Mayo")}<br>Su signo es Tauro`;

      } else if ((dia >= 21 && mes == 5) || (dia <= 20 && mes == 6)) {
        mensaje = `Su fecha de nacimiento es:<br> ${dia} de ${(mes == 5 ? "Mayo" : "Junio")}<br>Su signo es Geminis`;

      } else if ((dia >= 21 && mes == 6) || (dia <= 22 && mes == 7)) {
        mensaje = `Su fecha de nacimiento es:<br> ${dia} de ${(mes == 6 ? "Junio" : "Julio")}<br>Su signo Cancer`;

      } else if ((dia >= 23 && mes == 7) || (dia <= 22 && mes == 8)) {
        mensaje = `Su fecha de nacimiento es:<br> ${dia} de ${(mes == 7 ? "Julio" : "Agosto")}<br>Su signo es Leo`;

      } else if ((dia >= 23 && mes == 8) || (dia <= 22 && mes == 9)) {
        mensaje = `Su fecha de nacimiento es:<br> ${dia} de ${(mes == 8 ? "Agosto" : "Septiembre")}<br>Su signo es Virgo`;

      } else if ((dia >= 23 && mes == 9) || (dia <= 22 && mes == 10)) {
        mensaje = `Su fecha de nacimiento es:<br> ${dia} de ${(mes == 9 ? "Septiembre" : "Octubre")}<br>Su signo es Libra`;

      } else if ((dia >= 23 && mes == 10) || (dia <= 21 && mes == 11)) {
        mensaje = `Su fecha de nacimiento es:<br> ${dia} de ${(mes == 10 ? "Octubre" : "Noviembre")}<br>Su signo es Escorpio`;

      } else if ((dia >= 22 && mes == 11) || (dia <= 21 && mes == 12)) {
        mensaje = `Su fecha de nacimiento es:<br> ${dia} de ${(mes == 11 ? "Noviembre" : "Diciembre")}<br>Su signo es Sagitario`;

      } else if ((dia >= 22 && mes == 12) || (dia <= 19 && mes == 1)) {
        mensaje = `Su fecha de nacimiento es:<br> ${dia} de ${(mes == 12 ? "Diciembre" : "Enero")}<br>Su signo es Capricornio`;

      } else if ((dia >= 20 && mes == 1) || (dia <= 18 && mes == 2)) {
        mensaje = `Su fecha de nacimiento es:<br> ${dia} de ${(mes == 1 ? "Enero" : "Febrero")}<br>Su signo es Acuario`;

      } else if ((dia >= 19 && mes == 2) || (dia <= 20 && mes == 3)) {
        mensaje = `Su fecha de nacimiento es:<br> ${dia} de ${(mes == 2 ? "Febrero" : "Marzo")}<br>Su signo es Piscis`;

      }
    } else {
      mensaje = "Verifique la fecha de su nacimiento";
    }

    /* !SECTION fin codigo tras verificacion de formulario */
  } else {
    mensaje = `Ingrese un número válido en el formulario `;
  }

  const resultadoDiv = document.getElementById("resultadoJavascript");
  resultadoDiv.innerHTML = `${mensaje}`;
}

